polygon-digital-carbon/src/CreditManager.ts

```diff
@@ -0,0 +1,38 @@
+import { log } from '@graphprotocol/graph-ts'
+import {
+  CreditAdded as CreditAddedEvent,
+} from '../generated/CreditManager/CreditManager'
+import { CarbonProject } from '../generated/schema'
+
+export function handleCreditAdded(event: CreditAddedEvent): void {
+  log.info('handleProjectAdded fired {}-{}', [event.params.tokenAddress.toHexString(), event.params.tokenId.toString()])
+
+  const id = event.params.projectId
+  let project = CarbonProject.load(id)
+  if (project == null) {
+    project = new CarbonProject(id)
+  }
+    // Need to convert registry to the values we have previsouly hard coded in CarbonProject.ts
+    let registry = ''
+    const registryEvent = event.params.projectId.split('-')[0]
+    if (registryEvent == 'PURO') {
+      registry = 'PURO_EARTH'
```

Comment:
Just a note it'd be good to align these w/ marketplace subgraph if possible in the base version